central/policycategory/datastore/datastore_impl.go

```diff
@@ -54,6 +57,25 @@ func (ds *datastoreImpl) buildIndex() error {
 	return ds.indexer.AddPolicyCategories(categories)
 }
 
+func (ds *datastoreImpl) GetPolicyCategoriesForPolicy(ctx context.Context, policyID string) ([]*storage.PolicyCategory, error) {
+	ds.categoryMutex.Lock()
+	defer ds.categoryMutex.Unlock()
+
+	edges, err := ds.policyCategoryEdgeDS.SearchRawEdges(ctx, searchPkg.NewQueryBuilder().AddStrings(searchPkg.PolicyID, policyID).ProtoQuery())
```

Comment:
why search the join table? You should just be able to policyCategoryDS.Search(query) which would do a join for you.

Reply:
based on slack discussion, the suggestion ^ should work since the search will run on the edge and linked tables if the search term (in this case policy id) is not found in the category table.